cli/create-tauri-app/bin/create-tauri-app.js

```diff
@@ -168,81 +151,61 @@ async function runInit(argv, config = {}) {
   };
 
   if (recipe !== undefined) {
-    recipeSelection = recipe.shortName;
     buildConfig = recipe.configUpdate(buildConfig);
   }
 
   const directory = argv.d || process.cwd();
-
-  init({
-    directory,
-    force: argv.f || null,
-    logging: argv.l || null,
-    tauriPath: argv.t || null,
-    appName: appName || argv.A || null,
-    customConfig: merge(configOptions, {
-      build: buildConfig,
-      tauri: {
-        window: {
-          title: argv.W,
-        },
-      },
-    }),
-  });
-
-  const {
-    installDependencies,
-  } = require("../../tauri.js/dist/api/dependency-manager");
-  await installDependencies();
-
-  if (recipe !== undefined) {
-    const {
-      installRecipeDependencies,
-      runRecipePostConfig,
-    } = require("../../tauri.js/dist/api/recipes/install");
-
-    await installRecipeDependencies(recipe, directory);
-    await runRecipePostConfig(recipe, directory);
+  const cfg = {
+    ...buildConfig,
+    appName: appName || argv.A,
+    windowTitle: title || argv.w,
+  };
+  // note that our app directory is reliant on the appName and
+  // generally there are issues if the path has spaces (see Windows)
+  // future TODO prevent app names with spaces or escape here?
+  const appDirectory = join(directory, cfg.appName);
+
+  if (recipe.preInit) {
+    console.log("===== running initial command(s) =====");
+    await recipe.preInit({ cwd: directory, cfg });
   }
-}
-
-module.exports = main;
 
-/* will merge these later
-
-const parseArgs = require('minimist')
-const argv = parseArgs(process.argv.slice(2));
-
-/**
- * @description This is the bootstrapper that in turn calls subsequent
- * Tauri Commands
- *
- * @param {Object} argv
-*
-const tauri = (args) => {
-const tauri = async (args) => {
-  if (args["h"] || args["help"]) {
-    require("./help.js")();
-    return false // do this for node consumers and tests
+  const initArgs = [
+    ["--app-name", cfg.appName],
+    ["--window-title", cfg.windowTitle],
+    ["--dist-dir", cfg.distDir],
+    ["--dev-path", cfg.devPath],
+  ].reduce((final, argSet) => {
+    if (argSet[1]) {
+      return final.concat([argSet[0], `\"${argSet[1]}\"`]);
+    } else {
+      return final;
+    }
+  }, []);
+
+  if (!argv.b) {
+    console.log("===== adding tauri =====");
+    await shell("yarn add tauri --dev", {
+      cwd: appDirectory,
+    });
```

Comment:
I am on Arch Linux, for some reason, this was not working for me. I had to make following changes to work. :/
```suggestion
    await shell("yarn add tauri --dev", [], {
      cwd: appDirectory,
      shell: true,
    });
```

Reply:
I updated some of this if you don't mind giving it another shot. I am _trying_ to avoid using shell partly for security reasons and partly because it opens us up to other oddities (and hoping execa can handle most of that for us).